File: 240306/두 선분/two-lines.cpp
#include <iostream>

using namespace std;

bool checkIsIntersected(int x1, int x2, int x3, int x4){
    if(x3 <= x1 && x1 <= x4){
        return true;
    }
    if(x3 <= x2 && x2 <= x4){
        return true;
    }
    if(x1 <= x3 && x3 <= x2){
        return true;
    }
    if(x1 <= x4 && x4 <= x2){
        return true;
    }
    return false;
}

int main() {
    int x1, x2, x3, x4;
    cin >> x1 >> x2 >> x3 >> x4;

    if(checkIsIntersected(x1,x2,x3,x4)){
        cout << "intersecting";
    }else{
        cout << "nonintersecting";
    }
    return 0;
}

File: 250303/두 선분/two-lines.java
import java.util.Scanner;

public class Main {

    private static class Solver {

        int x1;
        int x2;
        int x3;
        int x4;

        public Solver(
            int x1,
            int x2,
            int x3,
            int x4
        ) {
            this.x1 = x1;
            this.x2 = x2;
            this.x3 = x3;
            this.x4 = x4;
        }

        public void solve() {
            boolean intersect = isIntersect();
            printResult(intersect);
        }

        private void printResult(boolean intersect) {
            if (intersect) {
                System.out.println("intersecting");
                return;
            }
            System.out.println("nonintersecting");
        }

        private boolean isIntersect() {
            if (x2 < x3 || x4 < x1) {
                return false;
            }
            return true;
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int x1 = sc.nextInt();
        int x2 = sc.nextInt();
        int x3 = sc.nextInt();
        int x4 = sc.nextInt();

        new Solver(x1, x2, x3, x4).solve();
    }
}